qiita_ware/ebi.py

```diff
@@ -105,21 +74,36 @@ class EBISubmission(object):
     Parameters
     ----------
     preprocessed_data_id : str
+
+    Raises
+    ------
+    EBISumbissionError
+        If either the required columns for EBI and Qiita deploy are not
+        present in the sample and prep templates
     """
 
     valid_ebi_actions = ('ADD', 'VALIDATE', 'MODIFY')
     valid_ebi_submission_states = ('submitting', 'success')
+    valid_platforms = ['LS454', 'ILLUMINA', 'UNKNOWN']
+    xmlns_xsi = "http://www.w3.org/2001/XMLSchema-instance"
+    xsi_noNSL = "ftp://ftp.sra.ebi.ac.uk/meta/xsd/sra_1_3/SRA.%s.xsd"
+    experiment_library_fields = [
+        'library_strategy', 'library_source', 'library_selection',
+        'library_layout']
 
     def __init__(self, preprocessed_data_id, action):
         valid_ebi_actions = EBISubmission.valid_ebi_actions
         valid_ebi_submission_states = EBISubmission.valid_ebi_submission_states
+        error_msgs = []
 
-        # Step 1: variable setup and validations
         if action not in valid_ebi_actions:
-            raise ValueError("Not a valid action (%s): %s" % (
-                ', '.join(valid_ebi_actions), action))
+            error_msg = "Not a valid action (%s): %s" % (
```

Comment:
If you wouldn't mind, modifying would be great!

On (Sep-04-15|11:11), Antonio Gonzalez wrote:
> >          if action not in valid_ebi_actions:
> > -            raise ValueError("Not a valid action (%s): %s" % (
> > -                ', '.join(valid_ebi_actions), action))
> > +            error_msg = "Not a valid action (%s): %s" % (
> 
> I can modify if you feel strongly about it. Note that all the information is there.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/1464/files#r38778913